SalesforceCommerceCloud/pwa-kit
# @W-13666502 - Removed offline products in baskets

# Description
This PR fixes an issue when a customer who has offline (or deleted) products in their basket or an existing order will have poor experience in cart and checkout. Errors are emitted to console in all cases.

Solution

Since basket does not have any property to indicate a productItem is offline. To solve this, after getting product details for basket's product items to populate onto the UI, we compare the number of items in basket and the number of returned data from `getProducts`. If a product is offline, the `getProducts` will not return any data related to that product. Hence, we know unavailable product ids, and remove them from basket before allowing users to interact with the cart and proceed to checkout.
See demo video here: https://www.loom.com/share/af80b139be1d4f4ab38af60340d97182?sid=698a44c8-60bb-4621-b0aa-60b446064a5f



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Checkout this branch
- Add items to cart
- Go to business manager, find these items and mark them as offline
- Navigate to cart page, you should expect to see a prompt to remove those items
- Click Remove items
- Offline products are removed completely from cart


- For wishlist page, do the same process

- Go to order page, observe that the unavailable products are still shown in the detail page but has a tag saying product unavailable next to the title, and it is no longer a link directing to a PDP
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/f815e036-4347-443a-9737-1acc8d207313)


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
